src/sentry/db/postgres/roles.py

```diff
@@ -1,42 +1,8 @@
 from __future__ import annotations
 
-import contextlib
-import os
-import sys
-from collections import defaultdict
-from typing import MutableMapping
 
-from django.db.transaction import get_connection
-
-from sentry.silo.patches.silo_aware_transaction_patch import determine_using_by_silo_mode
-
-_fencing_counters: MutableMapping[str, int] = defaultdict(int)
-
-
-@contextlib.contextmanager
 def in_test_psql_role_override(role_name: str, using: str | None = None):
-    """
-    During test runs, the role of the current connection will be swapped with role_name, and then swapped
-    back to its original.  Has no effect in production.
-    """
-
-    if "pytest" not in sys.argv[0] or os.environ.get("DB", "postgres") != "postgres":
-        yield
-        return
-
-    using = determine_using_by_silo_mode(using)
-
-    # TODO(mark) Move this closer to other silo code.
-    _fencing_counters[using] += 1
+    # Deprecated shim for getsentry compatibility
+    from sentry.testutils.silo import unguarded_write
```

Comment:
does this need the `using = determine_using_by_silo_mode` logic for split db still?

Reply:
Nevermind, it is captured inside the call.